scala/private/phases/phase_test_environment.bzl

```diff
@@ -1,14 +1,28 @@
 # Adds testing.TestEnvironment provider if "env" attr is specified
 # https://bazel.build/rules/lib/testing#TestEnvironment
 
+load("@bazel_skylib//lib:versions.bzl", "versions")
+
+
 def phase_test_environment(ctx, p):
     test_env = ctx.attr.env
+    inherited_environment = ctx.attr.inherited_environment
 
-    if test_env:
+    if inherited_environment:
         return struct(
             external_providers = {
-                "TestingEnvironment": testing.TestEnvironment(test_env),
+                "TestingEnvironment": testing.TestEnvironment(
+                        test_env,
+                        inherited_environment
```

Comment:
In order to retain partial compatibility with Bazel < 5.2.0, you could instantiate `TestingEnvironment` without `inherited_environment` if `env_inherit` has not been set by the user and set it in a separate branch.

Reply:
Unless I am not following, I think this is what I just did?

Reply:
You are right, looks good!